edx_rest_framework_extensions/auth/jwt/authentication.py

```diff
@@ -60,43 +60,62 @@ def get_jwt_claim_mergeable_attributes(self):
         return get_setting('JWT_PAYLOAD_MERGEABLE_USER_ATTRIBUTES')
 
     def authenticate(self, request):
+        is_jwt_from_authorization_header = self._is_jwt_from_authorization_header(request)
+
         try:
             user_and_auth = super().authenticate(request)
 
-            # Unauthenticated, CSRF validation not required
+            # Unable to authenticate, proceed to next Authentication class
             if not user_and_auth:
-                return user_and_auth
+                return None
 
-            # Not using JWT cookies, CSRF validation not required
-            use_jwt_cookie_requested = request.META.get(USE_JWT_COOKIE_HEADER)
-            if not use_jwt_cookie_requested:
+            # JWT found in authorization header, CSRF validation not required
+            if is_jwt_from_authorization_header:
                 return user_and_auth
 
+            # TODO: Why do we need CSRF protection but the original class does not?
+            # Enforce CSRF for JWT Cookie (assumed if not using the authorization header)
             self.enforce_csrf(request)
 
             # CSRF passed validation with authenticated user
             return user_and_auth
 
-        except jwt.InvalidTokenError as token_error:
-            # Note: I think this case is not used, but will monitor the custom attribute to verify.
-            set_custom_attribute('jwt_auth_failed', 'InvalidTokenError:{}'.format(repr(token_error)))
```

Comment:
**Reviewer Note:** These `jwt_auth_failed` custom attributes were added so we can use monitoring to better ensure that the new, less extreme, exception handling will capture all the correct exception classes.